### ISSUE TITLE
Compilation failure with local function inline_

### ISSUE THREAD

**Issue by elnabo** [HaxeFoundation/haxe#6908]
Having a local function named `inline_` yield unexpected results.

```haxe
class Test {
    static function main() {
        function inline_ (i:Int) { trace(i); }
        inline_(25);
    }
}
```

https://try.haxe.org/#007B7
Produce `Invalid_argument("index out of bounds")`
While a successful compilation or at least an error message is expected.

If the function `inline_` is declared any other way, the compilation is successful.
Tested on Neko/Js

**Comment by pleclech** [comment]
Hello,

This is because inline_XXX is used internally by the compiler to declare an inline function named XXX so when the compiler try to extract the name of the function from "inline_" it has nothing to  get back.

try to name it inline_inline_ and you can call it as inline_(), but as a side effect the function will be inlined:

You can see yourself:
https://try.haxe.org/#2070a

Regards

**Comment by RealyUniqueName** [comment]
Fixed with #7425
